File: Exercism Codes/C++/Triangle/triangle.cpp
#include "triangle.h"
#include <stdexcept>

namespace triangle {
	flavor kind(double a, double b, double c) {
		if (a + b < c || a + c < b || b + c < a || a <= 0 || b <= 0 || c <= 0) throw std::domain_error("");
		if (a == b && b == c && a == c) return flavor::equilateral;
		if (a == b || b == c || a == c) return flavor::isosceles;
		return flavor::scalene;
	}
}

File: Exercism Codes/Python/triangle.py
def equilateral(sides):
    return len(set(sides)) == 1 and inequality(sides)


def isosceles(sides):
    return len(set(sides)) in [1, 2] and inequality(sides)


def scalene(sides):
    return len(set(sides)) == 3 and inequality(sides)


def inequality(s):
    return s[0] + s[1] >= s[2] and s[1] + s[2] >= s[0] and s[0] + s[2] >= s[1] and all(x > 0 for x in s)
